huaweicloud/services/ims/data_source_huaweicloud_images_images.go

```diff
@@ -0,0 +1,318 @@
+				Optional: true,
+			},
+
+			"architecture": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"x86", "arm",
+				}, false),
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"os_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"image_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"flavor_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"images": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     ImagesImageRefSchema(),
+			},
+		},
+	}
+}
+
+func ImagesImageRefSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"visibility": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"container_format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"disk_format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"min_disk_gb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"min_ram_mb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protected": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"image_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"checksum": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"file": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"schema": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size_bytes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// dataSourceImagesImagesRead performs the image lookup.
+func dataSourceImagesImagesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	imageClient, err := config.ImageV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud image client: %s", err)
+	}
+
+	imageType := d.Get("visibility").(string)
```

Comment:
why do you use `visibility` as the image type, and use  imageType as virtaulEnvType?